solr/solr-ref-guide/src/reindexing.adoc

```diff
@@ -0,0 +1,185 @@
+
+It's possible to define separate analysis chains for indexing and query events, or you can define a single chain
+that is applied to both event types.
+
+If you change the analysis chain that applies to indexing events, it is strongly recommended that you reindex.
+This is because all of the changes that occur due to the chain configuration are applied to documents as they are
+being indexed, and only reindexing will allow your changes to take effect on documents.
+
+While not reindexing after analyzer changes is not required, be aware that not reindexing can cause unexpected
+query results in many cases.
+
+For example, if you indexed a number of documents and then decide you'd like to use the `LowerCaseTokenizerFactory`
+to ensure all text is converted to lower case, you will have a mix of entries in the field: some in their original
+case ("iPhone"), and newer documents in all lower-case ("iphone"). If you do not reindex the original set of documents,
+a query such as "iphone" will not match documents with "iPhone", because the schema rules enforce lower case on the
+query, but that's not what is in the index.
+
+The only time you do not have to reindex when changing a field type's analysis chain is when the changes impact
+queries *only* (and you know that you do not need to make corresponding changes to the index analysis).
+
+=== Solrconfig Changes
+
+Only one parameter change to Solr's `solrconfig.xml` requires reindexing. That parameter is the `luceneMatchVersion`,
+which controls the compatibility of Solr with Lucene changes. Since this parameter can change the rules for analysis behind the scenes, it's always recommended to reindex when changing this value. Usually, however, this is only changed in conjunction with a major upgrade.
+
+However, if you make a change to Solr's <<update-request-processors.adoc#update-request-processors,Update Request Processors>>, it's generally because you want to change something about how _update requests_ (documents) are _processed_ (indexed). In this case, you can decide based on the change if you want to reindex your documents to implement the changes you've made.
+
+Similarly, if you change the `codecFactory` parameter in `solrconfig.xml`, it is again strongly recommended that you
+plan to reindex your documents to avoid unintended behavior.
+
+== Upgrades
+
+When upgrading between major versions (for example, from a 7.x release to 8.0 or 8.x), a best practice
+is to always reindex your data.
+The reason for this is that subtle changes may occur in default field type definitions or the underlying code.
+
+[NOTE]
+If you have *not* changed your schema as part of an upgrade from one minor release to another (such as, from 7.x
+to 7.x), you can often get away with reindexing your documents without first deleting your index. However, when
+upgrading to a major release, you should plan to reindex your documents because of the likelihood of changes that break back-compatibility.
```

Comment:
This note is not clear to me. It seems to say that for minor version upgrades you can "reindex without deleting your index", but for major version upgrades "you should reindex". Is this second part supposed to be "reindex by first deleting your data/to a new collection"?

Reply:
I think I was making too subtle of a point - it's too awkwardly phrased to be clear. Rewording it to just say you usually don't have to reindex for minor version upgrades as long as you've made no changes that would usually trigger a need to reindex.